### ISSUE TITLE
Cannot use intrinsic function result as value of DefaultAuthorizer for AWS::Serverless::Api

### ISSUE THREAD

**Issue by greendavegreen** [awslabs/serverless-application-model#1138]
**Description:**

Cannot use intrinsic function as value of DefaultAuthorizer when attempting to parametize a template deploying AWS::Serverless::Api

**Steps to reproduce the issue:**
1. using SAM CLI, version 0.21.0
2. using a lambda authorizer attached to an API such as in the SAM example https://github.com/awslabs/serverless-application-model/tree/master/examples/2016-10-31/api_lambda_request_auth

This validates with SAM Validate:
```yaml
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31

Resources:

  MyApi:
    Type: AWS::Serverless::Api
    Properties:
      EndpointConfiguration: REGIONAL
      StageName: Prod
      Auth:
        DefaultAuthorizer: SAMTokenAuthorizer
        Authorizers:
          SAMTokenAuthorizer:
            FunctionArn: !GetAtt SamAuthFunction.Arn

  SamAuthFunction:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: ./auth_src
      Handler: authorizer.handler
      Runtime: nodejs8.10
```
3. attempt to make the choice of default Authorizer controlled by a parameter to the SAM template.  Add a string parameter and attempt to use !Ref or !Sub as value of DefaultAuthorizer.

This template **DOES NOT VALIDATE**:
```yaml
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31

Parameters:
  AuthType:
    Type: String
    Default: GuestTokenAuthorizer
    AllowedValues:
      - SAMTokenAuthorizer
      - GuestTokenAuthorizer

Resources:

  MyApi:
    Type: AWS::Serverless::Api
    Properties:
      EndpointConfiguration: REGIONAL
      StageName: Prod
      Auth:
        DefaultAuthorizer: !Ref AuthType
        Authorizers:
          SAMTokenAuthorizer:
            FunctionArn: !GetAtt SamAuthFunction.Arn
          GuestTokenAuthorizer:
            FunctionArn: !GetAtt GuestAuthInjector.Arn

  SamAuthFunction:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: ./auth_src
      Handler: authorizer.handler
      Runtime: nodejs8.10

  GuestAuthInjector:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: ./auth_src
      Handler: guest_auth.handler
      Runtime: nodejs8.10
```

**Observed result:**

I have been unable to find anything other than the string literal name of a declared Authorizer which can be used as the value for DefaultAuthorizer.

**Expected result:**

In order to follow SAM best practices and deploy a stack to different environments using parameters to customize behavior in dev/test/staging/prod, the Authorizer should accept values that can come from intrinsic functions.  I would like to use FindInMap to select values for authorizer based upon stack.

This also doesn't currently validate, but is the structure of what I would like to own (and I believe the spec intends to be legal)
```yaml
Parameters:
  EnvType:
    Type: String
    Default: test
    AllowedValues:
      - test
      - prod

Mappings:
  Authorizers:
    test:
      name: GuestTokenAuthozer
    prod:
      name: SAMTokenAuthorizer

Resources:

  MyApi:
    Type: AWS::Serverless::Api
    Properties:
      EndpointConfiguration: REGIONAL
      StageName: Prod
      Auth:
        DefaultAuthorizer: !FindInMap [Authorizers, !Ref EnvType, name]  
        Authorizers:
          SAMTokenAuthorizer:
            FunctionArn: !GetAtt SamAuthFunction.Arn
          GuestTokenAuthorizer:
            FunctionArn: !GetAtt GuestAuthInjector.Arn


```

**Comment by praneetap** [comment]
@greendavegreen Thanks for bringing up this issue. We do not (cannot easily) mix CFN resources and swagger extensions in a way that would consistently work. Since DefaultAuthorizers are added to the Api swagger, we do not allow intrinsics on the field.

**Comment by praneetap** [comment]
Closed as a duplicate of #1050
